dali/python/nvidia/dali/plugin/base_iterator.py

```diff
@@ -72,9 +72,15 @@ class _DaliBaseIterator(object):
                 samples that iterator needs to deal with. It allows `last_batch_policy` to be
                 PARTIAL or DROP, if FILL is used it is changed to PARTIAL. Sets `last_batch_padded`
                 accordingly to the reader's configuration (`pad_last_batch` reader argument)
-    auto_reset : bool, optional, default = False
-                Whether the iterator resets itself for the next epoch
-                or it requires reset() to be called separately.
+    auto_reset : string or bool, optional, default = False
+                Whether the iterator resets itself for the next epoch or it requires reset() to be called separately.
+
+                It can be one of the following values:
+
+                * ``"no"``, ``False`` or ``None`` - at the end of epoch StopIteration is raised and reset() needs to be called
+                * ``"yes"`` or ``"True"``- at the end of epoch StopIteration is raised but reset() is called internally automatically
+                * ``"silent"`` - data is returned infinitely without raising StopIteration; reset() is silently called internally
```

Comment:
1. As a food or thought: with this options extended I'd be very happy to see `cycle` parameter of external source deprecated.

2. In external source we call this mode `quiet` not `silent`.